addons/web/static/tests/views/fields/properties_field_tests.js

```diff
@@ -1400,4 +1481,318 @@ QUnit.module("Fields", (hooks) => {
+        await makePropertiesGroupView([false, false, false, false]);
+        await toggleSeparator("property_1", true);
+        assert.deepEqual(getGroups(), [
+            [
+                ["PROPERTY 1", "0600000000000000"],
+                ["Property 2", "property_2"],
+                ["Property 4", "property_4"],
+            ],
+            [
+                ["", "0600000000000000"],
+                ["Property 3", "property_3"],
+            ],
+        ]);
+
+        // fold the group
+        await click(
+            target,
+            ".o_field_properties .o_inner_group:first-child .o_field_property_label[property-name='0600000000000000']"
+        );
+        assert.deepEqual(getGroups(), [
+            [["PROPERTY 1", "0600000000000000"]],
+            [["", "0600000000000000"]],
+        ]);
+        await click(
+            target,
+            ".o_field_properties .o_inner_group:first-child .o_field_property_label[property-name='0600000000000000']"
+        );
+
+        await toggleSeparator("property_3", true);
+        assert.deepEqual(getGroups(), [
+            [
+                ["PROPERTY 1", "0600000000000000"],
+                ["Property 2", "property_2"],
+            ],
+            [
+                ["PROPERTY 3", "1200000000000000"],
+                ["Property 4", "property_4"],
+            ],
+        ]);
+
+        // fold the left group
+        await click(target, ".o_field_property_label[property-name='0600000000000000']");
+        assert.deepEqual(getGroups(), [
+            [["PROPERTY 1", "0600000000000000"]],
+            [
+                ["PROPERTY 3", "1200000000000000"],
+                ["Property 4", "property_4"],
+            ],
+        ]);
+        await click(target, ".o_field_property_label[property-name='0600000000000000']");
+
+        // create 3 new properties
+        await click(target, ".o_field_property_add button");
+        await click(target, ".o_field_property_add button");
+        await click(target, ".o_field_property_add button");
+        await nextTick();
+        await closePopover(target);
+        assert.deepEqual(getGroups(), [
+            [
+                ["PROPERTY 1", "0600000000000000"],
+                ["Property 2", "property_2"],
+            ],
+            [
+                ["PROPERTY 3", "1200000000000000"],
+                ["Property 4", "property_4"],
+                ["Property 5", "1b00000000000000"],
+                ["Property 6", "2200000000000000"],
+                ["Property 7", "2b00000000000000"],
+            ],
+        ]);
+
+        // Property 3 is not a separator anymore, should split in columns
+        await toggleSeparator("1200000000000000", false);
+        assert.deepEqual(getGroups(), [
+            [
+                ["PROPERTY 1", "0600000000000000"],
+                ["Property 2", "property_2"],
+                ["Property 4", "property_4"],
+                ["Property 6", "2200000000000000"],
```

Comment:
There was a big discussion about that behavior with FP and LNA, and the conclusion was to do the simplest thing from the code POV

Here we come from "1 and 3 are separators" to "only 1 is a separator", which is the same as "there's no separator" because the first element is the only separator

So by default (no separator) each item will be in it's position modulo the number of columns if that make sens (so if you have 2 columns, without separator the even and odd position will have their own column)

Hope I'm not too confusing, it's hard to explain don't hesitate if I'm not clear

Reply:
Ok I see, since you are removing 3 as separator, technically there is no separation since they need to go by 2 in order for them to work, so the columns default to the modulo `col_nbr` position. 
I think I understand what's happening, thanks for the explanation.

But this raises another question, what happens if you have 3 separator instead of 2 ? Or maybe this case does not exist ?

Reply:
> technically there is no separation since they need to go by 2 in order for them to work

Yes exactly :) 

> what happens if you have 3 separators instead of 2 ?

Good question, 3 separator means 3 or 4 groups (depends if the first element is a separator or not)


Lets' imagine we have 3 groups (first property is one of the 3 separators)

The group will be splitted like
```
A B
C
```

So the properties in those groups will looks like
```
a1  b1
a2  b2
    b3

c1
c2
c3
```
(alignement, etc are like normal fields in `<group/>`)

Reply:
But wouldn't the C group use the default behavior since they are not comprised between 2 separators:
```
a1  b1
a2  b2
    b3

c1  c3
c2
c4 (if we add a new property)
```

Reply:
no, it will only use the left column (the simplest possible behavior)